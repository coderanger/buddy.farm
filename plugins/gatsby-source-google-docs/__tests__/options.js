const documentLinks = require("./documents/links.json")
const documentTexts = require("./documents/texts.json")
const documentImages = require("./documents/images.json")
const documentFootnotes = require("./documents/footnotes.json")
const documentLists = require("./documents/lists.json")
const documentQuotes = require("./documents/quotes.json")
const documentCodes = require("./documents/codes.json")
const documentTables = require("./documents/tables.json")
const {GoogleDocument} = require("../utils/google-document")

test(`"KeepDefaultStyle" option`, () => {
  const options = {keepDefaultStyle: true}
  const googleDocument = new GoogleDocument({document: documentTexts, options})
  expect(googleDocument.toMarkdown()).toMatchSnapshot()
})

test(`"DemoteHeading" option enabled`, () => {
  const options = {demoteHeadings: true}
  const googleDocument = new GoogleDocument({document: documentTexts, options})
  expect(googleDocument.toMarkdown()).toMatchSnapshot()
})

test(`"DemoteHeading" option disabled`, () => {
  const options = {demoteHeadings: false}
  const googleDocument = new GoogleDocument({document: documentTexts, options})
  expect(googleDocument.toMarkdown()).toMatchSnapshot()
})

test(`Crosslinks between documents`, () => {
  const links = {
    [documentLinks.documentId]: "/relative-path",
    ["unknow"]: "/404",
  }

  const googleDocument = new GoogleDocument({
    document: documentLinks,
    links,
  })
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip headings`, () => {
  const options = {
    skipHeadings: true,
  }
  const googleDocument = new GoogleDocument({document: documentTexts, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip images`, () => {
  const options = {
    skipImages: true,
  }
  const googleDocument = new GoogleDocument({document: documentImages, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip footnotes`, () => {
  const options = {
    skipFootnotes: true,
  }
  const googleDocument = new GoogleDocument({
    document: documentFootnotes,
    options,
  })
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip lists`, () => {
  const options = {
    skipLists: true,
  }
  const googleDocument = new GoogleDocument({document: documentLists, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip quotes`, () => {
  const options = {
    skipQuotes: true,
  }
  const googleDocument = new GoogleDocument({document: documentQuotes, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip codes`, () => {
  const options = {
    skipCodes: true,
  }
  const googleDocument = new GoogleDocument({document: documentCodes, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})

test(`Skip tables`, () => {
  const options = {
    skipTables: true,
  }
  const googleDocument = new GoogleDocument({document: documentTables, options})
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})
