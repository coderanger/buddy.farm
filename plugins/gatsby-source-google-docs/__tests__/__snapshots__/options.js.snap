// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"DemoteHeading" option disabled 1`] = `
"# Title level 1

## Title level 2

### Title level 3

#### Title level 4

##### Title level 5

###### Title level 6

# Title

## Subtitle


**bold**


_italic_


<ins>underline</ins>


~~strikethrough~~


<sup>superscript</sup>


<sub>subscript</sub>


**_boldItalic_**


text with **space bold after** and _space italic before_.


		indented text


vertical<br/>indent


[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)


<span style='font-size:1.09em'>Text bigger</span>


<span style='color:rgb(106, 168, 79)'>Text Colored</span><span style='color:rgb(255, 0, 0)'>red</span><span style='color:rgb(0, 255, 0)'>green</span><span style='color:rgb(0, 0, 255)'>blue</span>


<span style='background-color:rgb(147, 196, 125)'>Text with background color</span><span style='background-color:rgb(255, 0, 0)'>red</span><span style='background-color:rgb(0, 255, 0)'>green</span><span style='background-color:rgb(0, 0, 255)'>blue</span>
"
`;

exports[`"DemoteHeading" option enabled 1`] = `
"## Title level 1

### Title level 2

#### Title level 3

##### Title level 4

###### Title level 5

###### Title level 6

## Title

### Subtitle


**bold**


_italic_


<ins>underline</ins>


~~strikethrough~~


<sup>superscript</sup>


<sub>subscript</sub>


**_boldItalic_**


text with **space bold after** and _space italic before_.


		indented text


vertical<br/>indent


[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)


<span style='font-size:1.09em'>Text bigger</span>


<span style='color:rgb(106, 168, 79)'>Text Colored</span><span style='color:rgb(255, 0, 0)'>red</span><span style='color:rgb(0, 255, 0)'>green</span><span style='color:rgb(0, 0, 255)'>blue</span>


<span style='background-color:rgb(147, 196, 125)'>Text with background color</span><span style='background-color:rgb(255, 0, 0)'>red</span><span style='background-color:rgb(0, 255, 0)'>green</span><span style='background-color:rgb(0, 0, 255)'>blue</span>
"
`;

exports[`"KeepDefaultStyle" option 1`] = `
"## <span style='font-size:1.45em;color:rgb(28, 69, 135)'>**Title level 1**</span>

### <span style='font-size:1.18em;color:rgb(61, 133, 198)'>**Title level 2**</span>

#### <span style='font-size:1.09em;color:rgb(109, 158, 235)'>**Title level 3**</span>

##### <span style='color:rgb(102, 102, 102)'><ins>Title level 4</ins></span>

###### <span style='color:rgb(102, 102, 102)'>Title level 5</span>

###### <span style='color:rgb(102, 102, 102)'>_Title level 6_</span>

## <span style='font-size:1.91em'>Title</span>

### <span style='font-size:1.18em;color:rgb(102, 102, 102)'>_Subtitle_</span>


**bold**


_italic_


<ins>underline</ins>


~~strikethrough~~


<sup>superscript</sup>


<sub>subscript</sub>


**_boldItalic_**


text with **space bold after** and _space italic before_.


		indented text


vertical<br/>indent


[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)


<span style='font-size:1.09em'>Text bigger</span>


<span style='color:rgb(106, 168, 79)'>Text Colored</span><span style='color:rgb(255, 0, 0)'>red</span><span style='color:rgb(0, 255, 0)'>green</span><span style='color:rgb(0, 0, 255)'>blue</span>


<span style='background-color:rgb(147, 196, 125)'>Text with background color</span><span style='background-color:rgb(255, 0, 0)'>red</span><span style='background-color:rgb(0, 255, 0)'>green</span><span style='background-color:rgb(0, 0, 255)'>blue</span>
"
`;

exports[`Crosslinks between documents 1`] = `
"
[to self](/relative-path)


[to self with user id](/relative-path)


[to self with edit](/relative-path)


[to self with preview](/relative-path)


[unknown url](https://docs.google.com/document/d/unknown)
"
`;

exports[`Skip codes 1`] = `
"
Type gatsby-source-google-docs-token to generate a token
"
`;

exports[`Skip footnotes 1`] = `
"
Text with a footnote and an other one
"
`;

exports[`Skip headings 1`] = `
"
**bold**


_italic_


<ins>underline</ins>


~~strikethrough~~


<sup>superscript</sup>


<sub>subscript</sub>


**_boldItalic_**


text with **space bold after** and _space italic before_.


		indented text


vertical<br/>indent


[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)


<span style='font-size:1.09em'>Text bigger</span>


<span style='color:rgb(106, 168, 79)'>Text Colored</span><span style='color:rgb(255, 0, 0)'>red</span><span style='color:rgb(0, 255, 0)'>green</span><span style='color:rgb(0, 0, 255)'>blue</span>


<span style='background-color:rgb(147, 196, 125)'>Text with background color</span><span style='background-color:rgb(255, 0, 0)'>red</span><span style='background-color:rgb(0, 255, 0)'>green</span><span style='background-color:rgb(0, 0, 255)'>blue</span>
"
`;

exports[`Skip images 1`] = `""`;

exports[`Skip lists 1`] = `""`;

exports[`Skip quotes 1`] = `""`;

exports[`Skip tables 1`] = `""`;
